[Core] Support GPU provisioning with flex-start in GKE (#6125)
skypilot-org/skypilot: docs/source/reference/config.rst

        runtimeClassName: nvidia
    :ref:`kueue <config-yaml-kubernetes-kueue>`:
      :ref:`local_queue_name <config-yaml-kubernetes-kueue-local-queue-name>`: skypilot-local-queue
    :ref:`dws <config-yaml-kubernetes-dws>`:

>       mode: flex-start or flex-start-queued-provisioning

Author: Updated

Name of the `local queue <https://kueue.sigs.k8s.io/docs/concepts/local_queue/>`_ to use for SkyPilot jobs.

.. _config-yaml-kubernetes-dws:

``kubernetes.dws``
~~~~~~~~~~~~~~~~~~

GKE DWS configuration (optional).

Refer to :ref:`Using DWS on GKE <dws-on-gke>` for more details.

``kubernetes.dws.enabled``
~~~~~~~~~~~~~~~~~~~~~~~~~~

Whether to enable DWS (optional).

> When ``enabled`` is set to ``true``, SkyPilot will automatically enable the flex-start with queued provisioning mode if ``kubernetes.kueue.local_queue_name`` is set, otherwise it will use the flex-start mode.

Reviewer: ```suggestion
When ``enabled: true``, SkyPilot will automatically use DWS with flex-start mode. If ``kubernetes.kueue.local_queue_name`` is set, it will use flex-start with queued provisioning mode.
```

Name of the `local queue <https://kueue.sigs.k8s.io/docs/concepts/local_queue/>`_ to use for SkyPilot jobs.

.. _config-yaml-kubernetes-dws:

> ``kubernetes.dws``

Reviewer 2: This option should only be allowed when the `autoscaler` is set to `gke`? We should have the check?

Author: Added check.

~~~~~~~~~~~~~~~~~~

> GKE DWS configuration (optional).

Reviewer 2: We should link to the `reservations.rst` section here for people wanting to see more details.

Author: Updated

``kubernetes.dws.mode``
~~~~~~~~~~~~~~~~~~~~~~~

Mode of the DWS (optional), see `GKE DWS documentation <https://cloud.google.com/kubernetes-engine/docs/concepts/dws>`_ for more details.

Can be one of:

- ``flex-start``: Flex start mode. If ``kubernetes.kueue.local_queue_name`` is set, this mode is automatically set to ``flex-start-queued-provisioning``.
- ``flex-start-queued-provisioning``: Flex start with queued provisioning mode.

> Default: ``null``.

Reviewer 2: What does it mean by Default: ``null`` for an user, considering you mention that it can only be one of the two options above?
